### PR TITLE
Fixed deleting/modifying

### PR DESCRIPTION
**Pull request by gaidds**
Now you can modify/delete products/bids only for your products/bids

### CODE REVIEW

File: routers/bid_router.py [hunk 1]
```diff
@@ -9,6 +9,8 @@
 
 @router.post('/add', response_model= BidDisplay)
 def add_bid(request: BidBase, db: Session = Depends(get_db), current_user: UserBase = Depends(get_current_user)):
+    # Set the bidder_id to the current user's id
+    request.bidder_id = current_user.id
```
Comment (Ashraf-Alshashaa) [inline routers/bid_router.py:13]: Whay we need auth here if any logged in user can add a bid

File: routers/bid_router.py [hunk 2]
```diff
@@ -9,6 +9,8 @@
 
 @router.post('/add', response_model= BidDisplay)
 def add_bid(request: BidBase, db: Session = Depends(get_db), current_user: UserBase = Depends(get_current_user)):
+    # Set the bidder_id to the current user's id
```
Comment (Ashraf-Alshashaa) [inline routers/bid_router.py:12]: pls without comments

File: routers/bid_router.py [hunk 3]
```diff
@@ -17,11 +19,35 @@ def get_all_bids(id: int = Query(..., alias='product_id'), db: Session = Depends
     return db_bid.get_all_bids(db, id)
 
 
+@router.get('/{id}', response_model=BidDisplay)
+def get_bid(id: int, db: Session = Depends(get_db)):
```
Comment (Ashraf-Alshashaa) [inline routers/bid_router.py:23]: do we need to get one bid?

File: routers/bid_router.py [hunk 4]
```diff
@@ -17,11 +19,35 @@ def get_all_bids(id: int = Query(..., alias='product_id'), db: Session = Depends
...
+    return db_bid.get_bid(db, id)
+
+
 @router.put('/status', response_model=BidDisplay)
-def change_bid_status(request: BidBase, bid_id: int,db: Session = Depends(get_db), current_user: UserBase = Depends(get_current_user)):
+def change_bid_status(request: BidBase, bid_id: int, db: Session = Depends(get_db), current_user: UserBase = Depends(get_current_user)):
+    # Fetch the bid to verify authorization
```
Comment (Ashraf-Alshashaa) [inline routers/bid_router.py:29]: comment

File: routers/bid_router.py [hunk 5]
```diff
@@ -17,11 +19,35 @@ def get_all_bids(id: int = Query(..., alias='product_id'), db: Session = Depends
...
+    bid = db_bid.get_bid(db, bid_id)
+    if not bid:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Bid not found")
+
+    # Check if the current user is the bidder
```
Comment (Ashraf-Alshashaa) [inline routers/bid_router.py:34]: comment

**Review by Laaad — APPROVED** [review]

File: routers/bid_router.py [hunk 6]
```diff
@@ -17,11 +19,35 @@ def get_all_bids(id: int = Query(..., alias='product_id'), db: Session = Depends
...
+
     return db_bid.change_bidding_status(db, bid_id, request)
 
 
 @router.delete('/delete/{id}')
-def delete_bid( id: int, db: Session =  Depends(get_db), current_user: UserBase = Depends(get_current_user)):
-    return db_bid.delete_bid(db, id)
\ No newline at end of file
+def delete_bid(id: int, db: Session = Depends(get_db), current_user: UserBase = Depends(get_current_user)):
+    # Fetch the bid to verify authorization
```
Comment (Ashraf-Alshashaa) [inline routers/bid_router.py:44]: comment

File: routers/bid_router.py [hunk 7]
```diff
@@ -17,11 +19,35 @@ def get_all_bids(id: int = Query(..., alias='product_id'), db: Session = Depends
...
+    bid = db_bid.get_bid(db, id)
+    if not bid:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Bid not found")
+
+    # Check if the current user is the bidder
```
Comment (Ashraf-Alshashaa) [inline routers/bid_router.py:49]: comment

File: routers/product_router.py [hunk 8]
```diff
@@ -8,6 +8,8 @@
 
 @router.post('/add', response_model=ProductDisplay)
 def add_product(request: ProductBase, db: Session = Depends(get_db), current_user: UserBase = Depends(get_current_user)):
+    # Set the seller_id to the current user's id
```
Comment (Ashraf-Alshashaa) [inline routers/product_router.py:11]: comment

File: routers/product_router.py [hunk 9]
```diff
@@ -18,10 +20,24 @@ def get_product(id: int, db: Session = Depends(get_db)):
 
 @router.put('/modify/{id}', response_model=ProductDisplay)
 def modify_product(id: int, request: ProductBase, db: Session = Depends(get_db), current_user: UserBase = Depends(get_current_user)):
+    # Fetch the product to verify ownership
```
Comment (Ashraf-Alshashaa) [inline routers/product_router.py:23]: comment

File: routers/product_router.py [hunk 10]
```diff
@@ -18,10 +20,24 @@ def get_product(id: int, db: Session = Depends(get_db)):
...
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Product not found")
+
+    if product.seller_id != current_user.id:
+        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail="Not authorized to modify this product")
     return db_product.modify_product(db, id, request)
 
 
 @router.delete('/delete/{id}')
 def delete_product(id: int, db: Session = Depends(get_db), current_user: UserBase = Depends(get_current_user)):
+    # Fetch the product to verify ownership
```
Comment (Ashraf-Alshashaa) [inline routers/product_router.py:35]: comment

File: routers/bid_router.py (hunk 3 above)
Comment (gaidds) [reply routers/bid_router.py:23]: yes, we need it later in code for authentication

File: routers/bid_router.py (hunk 4 above)
Comment (gaidds) [reply routers/bid_router.py:29]: ?

File: routers/bid_router.py (hunk 5 above)
Comment (gaidds) [reply routers/bid_router.py:34]: ?

File: routers/bid_router.py (hunk 1 above)
Comment (gaidds) [reply routers/bid_router.py:13]: There is no authentication. We just make sure that bid will have the bidder_id of user who made a bid

File: routers/bid_router.py (hunk 2 above)
Comment (gaidds) [reply routers/bid_router.py:12]: Why?

**Review by Ashraf-Alshashaa — APPROVED** [review]

**Pull request merged**